google_gke/variables.tf

```diff
@@ -1,101 +1,241 @@
+  description = "The GCE resource labels (a map of key/value pairs) to be applied to the cluster & other cluster-related resources. Merged with default labels (see locals.tf)."
+  type        = map(string)
+}
+
+variable "name" {
+  description = "Name of the cluster or application (required)."
+  type        = string
+}
+
+variable "project_id" {
+  default     = null
+  description = "The project ID to host the cluster in."
+  type        = string
+}
+
+variable "realm" {
+  description = "Name of infrastructure realm (e.g. prod or nonprod)."
+  type        = string
+
+  validation {
+    condition     = contains(["mgmt", "nonprod", "prod"], var.realm)
+    error_message = "Valid values for realm: mgmt, nonprod, prod."
+  }
+}
+
+variable "region" {
+  default     = "us-central1"
+  description = "Region where cluster & other regional resources should be provisioned. Defaults to us-central1."
   type        = string
-  description = "(Beta) The release channel of this cluster. Accepted values are `UNSPECIFIED`, `RAPID`, `REGULAR` and `STABLE`. Defaults to `UNSPECIFIED`."
+}
+
+variable "release_channel" {
   default     = "REGULAR"
+  description = "The release channel of this cluster. Accepted values are `UNSPECIFIED`, `RAPID`, `REGULAR` and `STABLE`. Defaults to `REGULAR`."
+  type        = string
 }
 
-variable "costcenter" {}
+variable "tags" {
+  default     = []
+  description = "The GCE resource tags (a list of strings) to be applied to the cluster & other cluster-related resources. Merged with default tags (see locals.tf)."
+  type        = list(string)
+}
 
-variable "project_id" {}
 
-variable "name" {}
+#
+# Cluster Maintenance Settings
+#
+variable "maintenance_exclusions" {
+  default     = []
+  description = "List of maintenance exclusions. A cluster can have up to three"
+  type        = list(object({ name = string, start_time = string, end_time = string }))
+}
 
-variable "network" {}
+variable "maintenance_start_time" {
+  default     = "21:00" # 1 or 2pm pacific, which is typically the low traffic point.
+  description = "Time window specified for daily or recurring maintenance operations in RFC3339 format"
+  type        = string
+}
 
-variable "subnetwork" {}
 
-variable "ip_range_pods" {
-  default = ""
+#
+# Cluster Networking
+#
+variable "shared_vpc_outputs" {
+  default     = null
+  description = "Sets networking-related variables based on a homegrown Shared VPC Terraform outputs data structure."
+  type = object({
+    ip_cidr_range = object({
+      master  = string
+      pod     = string
+      primary = string
+      service = string
+    })
+    network    = string
+    project_id = string
+    region     = string
+    secondary_ip_ranges = object({
+      pod = object({
+        ip_cidr_range = string
+        range_name    = string
+      })
+      service = object({
+        ip_cidr_range = string
+        range_name    = string
+      })
+    })
+    subnet_name = string
+    subnetwork  = string
```

Comment:
A heads up we are going to be exposing `subnetwork_id` since some gcp resources (like memcache) reference the network with the id. https://github.com/mozilla-it/global-platform-admin/pull/71